--- src/bvar/default_variables.cpp
+++ src/bvar/default_variables.cpp
@@ -442,19 +442,6 @@
         return false;
     }
     return true;
-#elif defined(OS_MACOSX)
-    // TODO(zhujiashun): get rchar, wchar, syscr, syscw, cancelled_write_bytes
-    // in MacOS.
-    memset(s, 0, sizeof(ProcIO));
-    static pid_t pid = getpid();
-    rusage_info_current rusage;
-    if (proc_pid_rusage(pid, RUSAGE_INFO_CURRENT, (void**)&rusage) != 0) {
-        PLOG(WARNING) << "Fail to proc_pid_rusage";
-        return false;
-    }
-    s->read_bytes = rusage.ri_diskio_bytesread;
-    s->write_bytes = rusage.ri_diskio_byteswritten;
-    return true;
 #else 
     return false;
 #endif


--- BUILD.bazel
+++ BUILD.bazel
@@ -51,7 +51,7 @@ LINKOPTS = [
     "-pthread",
     "-ldl",
 ] + select({
-    "@bazel_tools//tools/osx:darwin": [
+    "@platforms//os:macos": [
         "-framework CoreFoundation",
         "-framework CoreGraphics",
         "-framework CoreData",
@@ -217,7 +217,7 @@ BUTIL_SRCS = [
     "src/butil/recordio.cc",
     "src/butil/popen.cpp",
 ] + select({
-    "@bazel_tools//tools/osx:darwin": [
+    "@platforms//os:macos": [
         "src/butil/time/time_mac.cc",
         "src/butil/mac/scoped_mach_port.cc",
     ],
@@ -332,7 +332,7 @@ cc_library(
         "//bazel/config:brpc_with_glog": ["@com_github_google_glog//:glog"],
         "//conditions:default": [],
     }) + select({
-        "@bazel_tools//tools/osx:darwin": [":macos_lib"],
+        "@platforms//os:macos": [":macos_lib"],
         "//conditions:default": [],
     }) + select({
         "//bazel/config:brpc_with_boringssl": ["@boringssl//:ssl", "@boringssl//:crypto"],


